### ISSUE TITLE
[Tile] Expandable tiles: Adding guidance and rules for interactive elements

### ISSUE THREAD

**Issue by thyhmdo** [carbon-design-system/carbon-website#3499]
**Original issue:** https://github.com/carbon-design-system/carbon/issues/13030

From a discussion with Mike Gower, we want to add more guidance and rules to have a certain number of interactive elements or links within the actionable expandable tiles. Consider both with and without interactive elements expandable tiles. 

---

**Accepted criteria:**
From Gower - If you want users to be able to navigate tiles efficiently, do not put a whole bunch of click targets in tiles. Remember that actionable children within tiles add additional tab stops for keyboard users, which leads to much longer time on task. In an ideal world we’d recommend no more than one (two?) actions on a tile.

**Comment by laurenmrice** [comment]
- On the docket to discuss in the next CAG call.

**Comment by thyhmdo** [comment]
Looks like there are unknown issues of the number of elements and keyboard focus from Mike's perspective. Maybe next step is to do more research and see how the product teams are using them in their product. Also, consider the collapsed/expanded button position along with other action buttons (edit, overflow, trash,...)

**Comment by tay1orjones** [comment]
One additional thing here is that we recently updated the detection to include not only interactive elements, but also non-interactive elements with a role, such as StructuredList. https://github.com/carbon-design-system/carbon/pull/14239

This might be something to consider adding docs-wise.

**Comment by laurenmrice** [comment]
[add notes]

Sharing this at the next DSAG call.

**Comment by Laura-Marshall** [comment]
## Notes from DSAG on August 17, 2023


### Points covered on the call
- Problem occurs when there’s multiple interactive elements on a tile
- Problem with inputs hidden on expandable tiles (which is available in code)
- Two types of expandable tile, one with the input and one without (with different click interactions)
- Proposing two main types of tile: 
        - Base tiles (with and without interactive elements)
        - Clickable tiles (covering expandable, selectable)

**Initial problem space:**
- Some tiles that are clickable and some tiles that aren’t in the same place
- How do we indicate a difference visually between the clickable and non clickable tiles?

### Feedback and comments
_Is there any real product examples of text inputs in expandable tiles?_
- Text inputs within a drawable tile example
- There isn’t too many examples, but want to be accessibility forward in guidance
- Looking for examples

_How does these components impact the card? When does a tile become a card?_
- Its acting as behavioural guidance and a base component
- Offering don’ts 
- This should be less prescriptive, compared to card which is more example
- Tile is a component, card is a pattern

_How does a mix of borders look in something like a homepage?_
- Clear guidance needed to support this
- The mixing can work and be clean
- It may require refactoring

_Overflow menu plus a navigational tile is a main use case_

- When a static tile becomes clickable (in a dashboard), this would need different styling
- Only the tile would be operable (not the buttons on these tiles).
- We have a drag and drop pattern - looking to embed this into a component

_Impacts of expanding tiles when its within a grid_
_If its pushed down, it would push the whole row down
It would be good to have guidance around how this should work_ @lee-chase 
- Diana Tran has patterns for scaling the tiles

_Clickable rows in the data table often has different links within them showing specific information 
Can we apply this interaction to other places in the UI? 
Could a border be added to a data table? This might not work in this location._ (Kevin Camelo)
- How do we show that things are clickable on a touch device?
- Maybe Carbon can come up with a visual indicator to solve this
_How are other companies tackling this?_ 
_Context can be a big indicator on whether something is clickable_
https://www.nngroup.com/articles/clickable-elements/
 

### Next steps
- Produce a place for DSAG members to put example images in, and then ask for teams to add examples
    - Clickable tiles with actionable children on them
    - Multiple clickable actions within the tile
    - Are there examples where the expandable tile goes on top of the other tiles around it? - No examples yet from the group

**Comment by thyhmdo** [comment]
Reached out to couple stakeholders to acquire resources and links. Here is the [Figma file](https://www.figma.com/file/yHWOV93QHN8T8CBEJKw8N9/Tiles_V11_Carbon?type=design&node-id=0%3A1&mode=design&t=ZPYqWnoODK35EsQP-1) that includes those resources

Discussed with the design team, seem like next steps are:

- [ ] Gather and Synthesize the resources (probably a study case for Card pattern) 
- [ ] Stand firm on Tiles foundation by providing more Dos and Donts images to explain better the differences between card vs tile 
- [ ] In the future, providing the Card pattern would be beneficial to elevate Tile components

**Comment by Huixh111** [comment]
iiiii

**Comment by thyhmdo** [comment]
Closed this issue because it's duplicative and it's refine in this issue https://github.com/carbon-design-system/carbon-website/issues/3452
